config/config.go

```diff
@@ -804,6 +805,29 @@ type KubernetesSDConfig struct {
 	XXX map[string]interface{} `yaml:",inline"`
 }
 
+type KubernetesRole string
+
+const (
+	KubernetesRoleNode      = "node"
+	KubernetesRolePod       = "pod"
+	KubernetesRoleContainer = "container"
+	KubernetesRoleService   = "service"
+	KubernetesRoleEndpoint  = "endpoint"
+	KubernetesRoleAPIServer = "apiserver"
+)
+
+func (c *KubernetesRole) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	if err := unmarshal((*string)(c)); err != nil {
+		return err
+	}
+	switch *c {
+	case KubernetesRoleNode, KubernetesRolePod, KubernetesRoleContainer, KubernetesRoleService, KubernetesRoleEndpoint, KubernetesRoleAPIServer:
+		return nil
+	default:
+		return fmt.Errorf("Unknown Kubernetes SD role %q", c)
```

Comment:
What happens if no role is specified? A good error message will help the users who need to change their config.